src/NumericalAlgorithms/Interpolation/Actions/ElementInitInterpPoints.hpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "NumericalAlgorithms/Interpolation/Tags.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+class ConstGlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+namespace intrp {
+namespace Actions {
+
+/// \ingroup ActionsGroup
+/// \brief Adds interpolation point holders to the Element's DataBox.
+///
+/// This is for the case in which the points are time-independent.
+///
+/// This should be placed in the Initialization PDAL for DgElementArray.
+///
+/// Uses: nothing
+///
+/// DataBox changes:
+/// - Adds:
+///   - `intrp::Tags::InterpPointInfo<Metavariables>`
+/// - Removes: nothing
+/// - Modifies: nothing
+struct ElementInitInterpPoints {
+  template <typename DbTags, typename... InboxTags, typename ArrayIndex,
+            typename ActionList, typename Metavariables,
+            typename ParallelComponent,
+            Requires<not tmpl::list_contains_v<
+                DbTags, intrp::Tags::InterpPointInfo<Metavariables>>> = nullptr>
+  static auto apply(db::DataBox<DbTags>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    return std::make_tuple(
+        db::create_from<
+            db::RemoveTags<>,
+            db::AddSimpleTags<intrp::Tags::InterpPointInfo<Metavariables>>>(
+            std::move(box),
+            db::item_type<intrp::Tags::InterpPointInfo<Metavariables>>{}));
+  }
+
+  template <typename DbTags, typename... InboxTags, typename ArrayIndex,
```

Comment:
please combine these with if constexpr. (see #2200) and in the second case `ERROR`